enlace/programs/programs.php

```diff
@@ -74,9 +78,98 @@
             ?>
         </tr>
         <?php
-        $get_program_info = "SELECT * FROM Programs LEFT JOIN Institutions ON Host = Inst_ID ORDER BY Name";
+        $num_days_hidden = get_setting($NumDaysHiddenSetting);
+
+        $get_active_program_info = "SELECT Programs.Program_ID, Name, Institution_Name, MAX(Session_Names.End_Date) AS max_end_date FROM Programs " .
+            "LEFT JOIN Institutions ON Host = Inst_ID " .
+            "INNER JOIN Session_Names ON Session_Names.Program_ID = Programs.Program_ID " .
+            "           AND Session_Names.End_Date > NOW() - INTERVAL $num_days_hidden DAY " .
+            "GROUP BY Programs.Program_ID " .
+            "ORDER BY Name";
+
+        include "../include/dbconnopen.php";
+        $program_info = mysqli_query($cnnEnlace, $get_active_program_info);
+        while ($temp_program = mysqli_fetch_array($program_info)) {
+            ?><tr>
+                <td><a href="javascript:;" onclick="$.post(
+                                '../ajax/set_program_id.php',
+                                {
+                                    page: 'profile',
+                                    id: '<?php echo $temp_program['Program_ID']; ?>'
+                                },
+                        function(response) {
+                            window.location = 'profile.php';
+                        }
+                        ).fail(failAlert);"><?php echo $temp_program['Name']; ?></a>
+                </td>
+                <td>
+                    <?php echo $temp_program['Institution_Name']; ?>
+                </td>
+                <?php
+                //if an administrator
+            if ($USER->has_site_access($Enlace_id, $AdminAccess)) {
+                    //show delete area
+                    ?>
+                    <td style="text-align: center;">
+                        <a href="javascript:;" onclick="
+                                            if (confirm('ARE YOU SURE YOU WANT TO DELETE THIS PROGRAM?')) {
+                                                if (confirm('ARE YOU SURE YOU WANT TO DELETE THIS PROGRAM?\r\nNOTE: This will remove all session data as well.')) {

```

Comment:
Whoa.  We're offering Program deletion functionality?  Was that part of the issue?  How is it related to hiding of old sessions?

Reply:
Okay.  I did go look for corresponding removal ("-") lines elsewhere and didn't see them, but I could have missed it.  As long as it's just code movement, not code change, no worries.